### PR TITLE
Implement the new interface in AE2 for network bytes information.

### PR DESCRIPTION
**Pull request by MCTBL**

### CODE REVIEW

File: src/main/java/thaumicenergistics/common/inventory/HandlerItemEssentiaCell.java [hunk 1]
```diff
@@ -605,47 +607,49 @@ public List<IAspectStack> getStoredEssentia() {
...
-        return this.totalTypes;
-    }
+    // /**
+    // * Total number of types the cell can hold.
+    // *
+    // * @return
+    // */
+    // public int getTotalTypes() {
+    // return this.totalTypes;
+    // }
```
Comment (Alastors) [inline src/main/java/thaumicenergistics/common/inventory/HandlerItemEssentiaCell.java:622]: Why are these just commented out?

File: src/main/java/thaumicenergistics/common/inventory/HandlerItemEssentiaCell.java [hunk 2]
```diff
@@ -605,47 +607,49 @@ public List<IAspectStack> getStoredEssentia() {
...
-        }
-
-        // Return the count
-        return typeCount;
-    }
+    // /**
+    // * Returns how many types are used.
+    // *
+    // * @return
+    // */
```
Comment (Alastors) [inline src/main/java/thaumicenergistics/common/inventory/HandlerItemEssentiaCell.java:638]: Again?

**Review by Alastors — CHANGES_REQUESTED** [review]

File: src/main/java/thaumicenergistics/common/inventory/HandlerItemEssentiaCell.java (hunk 1 above)
Comment (MCTBL) [reply src/main/java/thaumicenergistics/common/inventory/HandlerItemEssentiaCell.java:622]: I make a new interface `ICellCacheRegistry` in AE2, that helps me to collect cell bytes and type info, and I accidentally use the same name in the interface. But return is long in the interface.

Comment (Alastors) [reply src/main/java/thaumicenergistics/common/inventory/HandlerItemEssentiaCell.java:622]: Yeah but you should terminate them if that's the case

Comment (MCTBL) [reply src/main/java/thaumicenergistics/common/inventory/HandlerItemEssentiaCell.java:622]: OK, I'm working on it.

**Comment by Cardinalstars** [comment]
Is this change in the wake of something changed in base AE2? If so a link would be helpful.

**Comment by MCTBL** [comment]
> Is this change in the wake of something changed in base AE2? If so a link would be helpful.

https://github.com/GTNewHorizons/Applied-Energistics-2-Unofficial/pull/419
I'm trying to add a byte information for ae network.

**Comment by Alastors** [comment]
I don't know enough about the AE2 codebase to knowingly review this past things such as the commented out code from earlier.

**Comment by MCTBL** [comment]
> I don't know enough about the AE2 codebase to knowingly review this in good conscious past things such as the commented out code from earlier.

I have not changed the `totalTypes` type, it still a int variable. I just change the method return type from int to long. And method `getUsedTypes` as the same.

**Review by Laiff — APPROVED** [review]

**Pull request merged**